--- a/xen/common/argo.c	2019-09-26 14:33:22.060786847 -0400
+++ b/xen/common/argo.c	2019-09-26 14:27:18.294107360 -0400
@@ -2263,10 +2263,12 @@
                                         src_id.domain_id);
     if ( !ring_info )
     {
+	/*
         gprintk(XENLOG_ERR,
                 "argo: vm%u connection refused, src (vm%u:%x) dst (vm%u:%x)\n",
                 current->domain->domain_id, src_id.domain_id, src_id.aport,
                 dst_addr->domain_id, dst_addr->aport);
+	*/
 
         ret = -ECONNREFUSED;
     }
